src/core/components/libp2p.js

```diff
@@ -25,10 +25,15 @@ module.exports = function libp2p (self) {
         self._libp2pNode = new Node(self._peerInfo, self._peerInfoBook, options)
 
         self._libp2pNode.on('peer:discovery', (peerInfo) => {
-          if (self.isOnline()) {
+          const dial = () => {
             self._peerInfoBook.put(peerInfo)
             self._libp2pNode.dial(peerInfo, () => {})
           }
+          if (self.isOnline()) {
+            dial()
+          } else {
+            self._libp2pNode.once('online', dial)
```

Comment:
Oops, I knew I had another PR to submit...
See https://github.com/libp2p/js-libp2p/pull/93